File: examples/Two_Planar_Near/planar_near.cpp
#include "setup.h"

int main(){
  // initializing simulation
  Ptr<SimulationPlanar> s = SimulationPlanar::instanceNew();
  s->addMaterial("GaAs", "GaAs.txt");
  s->addMaterial("Vacuum", "Vacuum.txt");
  s->addMaterial("PEC", "PEC.txt");

  // add layer from bottom to the top
  s->addLayer("PECBottom", 0, "PEC");
  s->addLayer("GaAsBottom", 1e-6, "GaAs");
  s->addLayer("VacGap", 1e-8, "Vacuum");
  s->addLayerCopy("GaAsTop", "GaAsBottom");
  s->addLayerCopy("PECTop", "PECBottom");

  // set source and probe
  s->setSourceLayer("GaAsBottom");
  s->setProbeLayer("VacGap");

  // set simulation
  s->setKParallelIntegral(10);
  s->optUseQuadgk();
  s->buildRCWA();
  s->setThread(4);
  s->integrateKParallel();
  s->saveToFile("test.txt");
  double* phi = s->getPhi();
  double* omega = s->getOmega();
  for(int i = 0; i < s->getNumOfOmega(); i++){
    std::cout << omega[i] << "\t" << phi[i] << std::endl;
  }
  return 0;
}

File: examples/Two_Planar_Near/planar_near.lua
-- this is a lua realization of the corresponding cpp file
s = SimulationPlanar.new()
s:AddMaterial("GaAs", "GaAs.txt")
s:AddMaterial("PEC", "PEC.txt")
s:AddMaterial("Vacuum", "Vacuum.txt")

s:AddLayer("PECBottom", 0, "PEC");
s:AddLayer("GaAsBottom", 1e-6, "GaAs");
s:AddLayer("VacGap", 1e-8, "Vacuum");
s:AddLayerCopy("GaAsTop", "GaAsBottom");
s:AddLayerCopy("PECTop", "PECBottom");

s:SetSourceLayer("GaAsBottom");
s:SetProbeLayer("VacGap");
s:OptUseQuadgk();
s:SetKParallelIntegral(10);
s:SetThread(4);

s:BuildRCWA();
s:IntegrateKParallel();
s:SaveToFile("test_lua.txt");

numOfOmega = s:GetNumOfOmega();
phi = s:GetPhi();
omega = s:GetOmega();
for i = 1,s:GetNumOfOmega(), 1 do
  print(string.format("%e", omega[i]).."\t"..string.format("%e", phi[i]));
end